martin/src/cog/source.rs

```diff
@@ -140,9 +162,33 @@ impl CogSource {
             CogError::ZoomOutOfRange(xyz.z, self.path.clone(), self.min_zoom, self.max_zoom)
         })?;
 
-        let bytes = image.get_tile(&mut decoder, xyz, self.nodata, &self.path)?;
-        Ok(bytes)
+        if self.web_friendly {
+            // just clip the image to get the tile in web mercator
+            let bytes = image.get_tile_webmercator(&mut decoder, xyz, self.nodata, &self.path)?;
+            Ok(bytes)
+        } else {
+            let bytes = image.get_tile(&mut decoder, xyz, self.nodata, &self.path)?;
+            Ok(bytes)
+        }
+    }
+}
+
+/// find a zoom level of google web mercator that is closest to the given resolution
+fn nearest_web_mercator_zoom(resolution: (f64, f64), tile_size: (u32, u32)) -> u8 {
+    let tile_width_in_model = resolution.0 * f64::from(tile_size.0);
+    let mut nearest_zoom = 0u8;
+    let mut min_diff = f64::INFINITY;
+
+    for google_zoom in 0..30 {
+        let tile_length = EARTH_CIRCUMFERENCE / f64::from(1_u32 << google_zoom);
+        let current_diff = (tile_width_in_model - tile_length).abs();
+
+        if current_diff < min_diff {
+            min_diff = current_diff;
+            nearest_zoom = google_zoom;
+        }
     }
```

Comment:
this is the only place where google is mentioned

```suggestion
/// Find the closest web mercator zoom level for the given resolution
fn nearest_web_mercator_zoom(resolution: (f64, f64), tile_size: (u32, u32)) -> u8 {
    let tile_width_in_model = resolution.0 * f64::from(tile_size.0);
    let mut nearest_zoom = 0u8;
    let mut min_diff = f64::INFINITY;

    for zoom in 0..30 {
        let tile_length = EARTH_CIRCUMFERENCE / f64::from(1_u32 << zoom);
        let current_diff = (tile_width_in_model - tile_length).abs();

        if current_diff < min_diff {
            min_diff = current_diff;
            nearest_zoom = zoom;
        }
    }
```